web/tests/stream_data.test.js

```diff
@@ -166,16 +162,21 @@ test("basics", () => {
     assert.ok(!stream_data.is_default_stream_id(social.stream_id));
     assert.ok(!stream_data.is_default_stream_id(999999));
 
-    assert.equal(stream_data.slug_to_name("2-social"), "social");
-    assert.equal(stream_data.slug_to_name("2-whatever"), "social");
-    assert.equal(stream_data.slug_to_name("2"), "social");
-
+    // "new" correct url formats
+    assert.equal(stream_data.slug_to_stream_id("2-social"), 2);
+    assert.equal(stream_data.slug_to_stream_id("2"), 2);
+    // we still get 2 because it's a valid stream id
+    assert.equal(stream_data.slug_to_stream_id("2-whatever"), 2);
+    // invalid stream id
+    assert.equal(stream_data.slug_to_stream_id("999-social"), undefined);
     // legacy
-    assert.equal(stream_data.slug_to_name("25-or-6-to-4"), "25-or-6-to-4");
```

Comment:
This is a valid thing that can appear in a legacy link, so it should continue to work. That means it should continue to resolve to a stream with this name — if such a stream exists.

That's the "311-help" example in the mobile tests:
https://github.com/zulip/zulip-mobile/blob/2217c858e207f9f092651dd853051843c3f04422/src/utils/__tests__/internalLinks-test.js#L333-L354

The "311-" and "311" examples are related edge cases that would also be good to test.

Reply:
Should `25-or-6-to-4` return `"25"` as a stream id? Or `"25-or-6-to-4"` as a full stream name because 25 isn't a valid stream id? I just switched to doing the former, but that will change this test case from how it was before. This is for an operand for the stream operator, and I think actually keeping it as a full string for cases when we don't recognize the id makes more sense?

Reply:
I think the "311" test case is `assert.equal(stream_data.slug_to_name("2"), "2");`

I can add a `assert.equal(stream_data.slug_to_name("2-"), "2");`

(these are both for valid stream ids)

Reply:
Yeah I agree! I was working on changing it something like that, I'll just push it and add some test cases to show what's going on.

Reply:
Followup thread below: https://github.com/zulip/zulip/pull/31299#discussion_r1765739302